src/main.rs

```diff
@@ -1,3 +1,113 @@
+
+    let about_menu = gio::MenuItem::new("About", "app.about");
+    menu.append_item(&about_menu);
+
+    // Connect the about menu entry click event.
+    let about = gio::SimpleAction::new("about", None);
+    let weak_window = window.downgrade();
+    about.connect_activate(move |_, _| {
+        let window = upgrade_weak!(weak_window);
+        let p = gtk::AboutDialog::new();
+        p.set_authors(&["Sebastian Dröge", "Guillaume Gomez"]);
+        p.set_website_label(Some("github repository"));
+        p.set_website(Some(
+            "https://github.com/sdroege/rustfest-rome18-gtk-gst-workshop",
+        ));
+        p.set_comments(Some("A video input reader written with gtk-rs and gstreamer-rs"));
+        p.set_copyright(Some("This is under MIT license"));
+        p.set_transient_for(Some(&window));
+        p.set_modal(true);
+        p.set_program_name("Rustfest GTK+GStreamer");
+        p.show_all();
+    });
+    let quit = gio::SimpleAction::new("quit", None);
+    let weak_window = window.downgrade();
+    quit.connect_activate(move |_, _| {
+        let window = upgrade_weak!(weak_window);
+        window.destroy();
+    });
+
+    application.add_action(&about);
+    application.add_action(&quit);
+
+    application.set_app_menu(&menu);
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(application);
+
+    window.set_title("Rustfest 2018 GTK+GStreamer");
+    window.set_border_width(5);
+    window.set_position(gtk::WindowPosition::Center);
+    window.set_default_size(350, 300);
+
+    window.connect_delete_event(move |win, _| {
+        win.destroy();
+        Inhibit(false)
+    });
+
+    let combo_box = gtk::ComboBoxText::new();
+    combo_box.append_text("<Pick a video input>");
+    combo_box.set_active(0);
+
+    // TODO: put different possible video inputs here. For now it's just for the show up
+    combo_box.append_text("Video input 3");
+    combo_box.append_text("Video input 1337");
+
+    combo_box.connect_property_active_notify(|cb| {
+        if let Some(active) = cb.get_active_text() {
+            if active != "<Pick a video input>" {
+                println!("new active: {}", active);
+            }
+        }
+    });
+
+    let vertical_layout = gtk::Box::new(gtk::Orientation::Vertical, 0);
+    vertical_layout.pack_start(&combo_box, false, true, 0);
+    let draw_area = gtk::DrawingArea::new();
+    vertical_layout.pack_start(&draw_area, true, true, 0);
+
+    window.add(&vertical_layout);
+
+    build_menu(application, &window);
+
+    window.show_all();
+}
+
 fn main() {
-    println!("Hello, world!");
+    let application = gtk::Application::new("com.github.rustfest",
+                                            gio::ApplicationFlags::empty())
+                                       .expect("Initialization failed...");
+
+    application.connect_startup(|app| {
+        build_ui(app);
+    });
+    application.connect_activate(|_| {});
```

Comment:
Because of:

> (.:91070): GLib-GIO-WARNING **: 08:58:30.931: Your application does not implement g_application_activate() and has no handlers connected to the 'activate' signal.  It should do one of these.

I've removed it anyway.